robot_smach_states/src/robot_smach_states/perception.py

```diff
@@ -91,25 +91,29 @@ def __init__(self, robot, entity, keep_following=False, waittime=0.0):
         ds.check_type(entity, Entity)
 
         State.__init__(self, locals(), outcomes=['succeeded', 'failed'])
+        self._robot = robot
+	self._entity = entity
+	self._keep_following = keep_following
+	self._waittime = waittime
 
-    def run(self, robot, entity, keep_following, waittime):
-        if keep_following:
+    def run(self):
+        if self._keep_following:
             rospy.logerr("Look at stuff: keep_following is obsolete")
 
-        if not entity:
+        if not self._entity:
             return 'failed'
 
         # Entities define their own frame, so there is no need to transform the pose to /map.
         # That would be equivalent to defining coordinates 0,0,0 in its own frame, so that is what we do here.
         # The added benefit is that the entity's frame actually moves because the entity is tracked.
         # This makes the head track the entity
-        center_point = VectorStamped(frame_id="/"+entity.id)
+        center_point = VectorStamped(frame_id="/"+self._entity.id)
 
-        center_point.vector.z(entity.shape.z_max)
+        center_point.vector.z(self._entity.shape.z_max)
 
         rospy.loginfo('Look at %s' % (repr(center_point)))
-        robot.head.look_at_point(center_point)
-        rospy.sleep(rospy.Duration(waittime))
+        self._robot.head.look_at_point(center_point)
+        rospy.sleep(rospy.Duration(self._waittime))
```

Comment:
These changes are an improvement and don't change the functionality, so should be kept.